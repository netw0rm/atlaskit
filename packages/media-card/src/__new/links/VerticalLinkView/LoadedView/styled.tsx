/* tslint:disable: variable-name */
import * as React from 'react';
import styled from 'styled-components';
import {minSquareCardDimensions, maxSquareCardDimensions} from '../../../../utils/cardDimensions';

// TODO: this needs to be composed inside the component (so we're not duplicating the width calcs)
// but can probably be reused across multiple components

const A = styled.a`
  display: block;
  min-width: ${minSquareCardDimensions.width}px;
  max-width: ${maxSquareCardDimensions.width}px;
  text-decoration: none;
  &:hover {
    text-decoration: none;
  }
`;

export interface AnchorProps {
  href?: string;
  onClick?: () => void;
}

export class Anchor extends React.Component<AnchorProps, {}> {

    handleClick = (event) => {
      event.preventDefault();
      const {onClick} = this.props;
      if (onClick) {
        onClick();
      }
    }

    render() {
      const {href, children} = this.props;
      return (
        <A className="link-wrapper" href={href} onClick={this.handleClick}>{children}</A>
      );
    }
  }
