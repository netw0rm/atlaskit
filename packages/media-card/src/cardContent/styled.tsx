/* tslint:disable:variable-name */
import styled from 'styled-components';
import { easeInOutCubic } from '../styles/easing';
import { fadeIn } from '../styles/animations';

export const LoadingWrapper = styled.div`
  color: #cfd4db;
  display: flex;
  height: 100%;
  justify-content: center;
  align-items: center;
`;

export const FadeinImage = styled.div`
  animation: ${fadeIn} .3s ${easeInOutCubic};
`;
