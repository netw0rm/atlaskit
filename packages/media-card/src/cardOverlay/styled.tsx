/* tslint:disable:variable-name */
import styled from 'styled-components';
import { akColorN70 } from '@atlaskit/util-shared-styles';

export const MoreBtn = styled.div`
  display: none;
  float: right;
  width: 36px;
  height: 26px;
  border-radius: 3px;

  &:hover {
    background-color: rgba(9, 30, 66, 0.06);
  }

  &.active {
    background-color: white !important;

    span {
      color: #253858;
    }
  }
`;

export const DeleteBtn = styled.div`
  display: none;
  float: right;
  width: 26px;
  height: 26px;
  border-radius: 3px;
  color: white;
  justify-content: center;

  &:hover {
    background-color: rgba(9, 30, 66, 0.06);
  }
`;

export const FileTypeIcon = styled.div`
  float: left;
  margin-right: 6px;
  position: relative;
  top: 1px;
  
  &.audio {
    color: #8777D9;
  }

  &.doc {
    color: #0065ff;
  }
  
  &.image {
    color: #ffc400;
  }

  &.video {
    color: #ff7143;
  }

  &.unknown {
    color: #3dc7dc;
  }

  span{
    width: 12px !important;
    height: 12px !important;
  }
`;

export const TickBox = styled.div`
  position: absolute;
  top: 8px;
  right: 8px;
  width: 14px;
  height: 14px;
  border: 2px solid white;
  border-radius: 20px;
  z-index: 20;
  display: none;
`;

export const Overlay = styled.div`
  width: 100%;
  height: 100%;
  background: transparent;
  position: absolute;
  top: 0;
  left: 0;
  border-radius: 3px;
  border: 2px solid transparent;

  &:hover, &.active {
    .top-row {
      .title {
        color: #0065FF;
      }
    }

    .left-column{
      opacity: 1;
      transform: translateY(0);
    }

    .bottom-row {
      .more-btn {
        display: flex;
        justify-content: center;
      }

      .delete-btn {
        display: flex;
      }
    }
  }
  
  &.video, &.doc, &.unknown {
    .file-type-icon {
      display: block;
    }
  }
  /* Image */
  &.image {
    .top-row {
      .title {
        color: white;
        font-size: 12px;
        visibility: hidden;
      }
    }

    .bottom-row {
      .file-type-icon {
        display: none;
      }

      .file-size {
        color: white;
        display: none;
      }

      .more-btn {
        display: none;

        &:hover {
          background-color: rgba(9, 30, 66, 0.20);
        }
      }

      .delete-btn {
        display: none;

        &:hover {
          background-color: rgba(9, 30, 66, 0.20);
        }
      }
    }

    &:hover, &.active {
      background-color: rgba(9, 30, 66, 0.5);

      .title {
        visibility: visible;
        transform: translateY(0);
        opacity: 1;
      }

      .file-type-icon {
        display: block;
      }

      .file-size {
        display: block;
      }

      .more-btn {
        display: flex;
        justify-content: center;
        color: white;
      }

      .delete-btn {
        display: flex;
      }
    }

    /* Selectable */
    &.selectable {
      &:hover, &.active {
        .tickbox {
          display: block;
        }
      }

      &.selected {
        border-color: #3384FF !important;

        .tickbox {
          display: flex !important;
          background-color: #3384FF !important;
          border-color: #3384FF !important;
          color: white;
        }
      }
    }

    /* In progress*/
    &.in-progress {
      .bottom-row {
        .file-type-icon {
          display: none;
        }

        .file-size {
          display: none;
        }
      }
    }
  }
  /* Error state */
  &.error {
    .top-row{
      overflow: visible;
    }
    &:hover, &.active {
      .top-row {
        .title {
          color: #091E42;
        }
      }
    }
  }

  .file-type-icon {
    display: none;
  }
`;

export const ErrorLine = styled.div`
  display: block;
  height: 24px;
  display: flex;
  align-items: center;
`;

export const LeftColumn = styled.div`
  width: 100%;
  display: table-cell;
  position: relative;
  box-sizing: border-box;
  vertical-align: middle;
  padding: 0 5px 0 5px;
  opacity: 0;
  transform: translateY(25px);
  transition: all .3s;
`;

export const TopRow = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  box-sizing: border-box;
  padding: 8px 8px 0 8px;
`;

export const BottomRow = styled.div`
  position: absolute;
  display: table;
  bottom: 0;
  left: 0;
  width: 100%;
  box-sizing: border-box;
  height: 28px;
  line-height: 26px;
  padding: 0 2px 0 2px;
`;

export const RightColumn = styled.div`
  display: table-cell;
`;

export const ErrorIconWrapper = styled.div`
  display: flex;
  color: #ff991f;
`;

export const ErrorMessage = styled.div`
  display: inline-block;
  vertical-align: middle;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: bold;
  color: ${akColorN70};
  font-size: 12px;
  line-height: 15px;
  overflow: hidden;
  max-width: ~"calc(100% - 24px)";
  text-overflow: ellipsis;
  white-space: nowrap;
`;

export const Retry = styled.div`
  position: absolute;
  box-sizing: border-box;
  padding: 0 7px 7px 0;
  bottom: 0;
  width: 100%;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: bold;
  color: #0065FF;
  font-size: 12px;
  line-height: 15px;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
`;

export const DropdownWrapper = styled.div`
  position: absolute;
  z-index: 100;
  left: calc(100% - 40px);
  top: 100%;
  display: block;
`;

export const TitleWrapper = styled.div`
  box-sizing: border-box;
  word-wrap: break-word;
  color: #091E42;
  font-size: 12px;
  line-height: 18px;
  transform: translateY(-25px);
  transition: all .3s;
  opacity: 0;
`;

export const FileSize = styled.div`
  float: left;
  font-size: 12px;
  color: #5E6C84;
`;

export const Metadata = styled.div`
  &.has-progress {
    display: none;
  }
`;
