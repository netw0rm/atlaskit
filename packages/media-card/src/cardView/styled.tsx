/* tslint:disable:variable-name */
import styled from 'styled-components';
import { Root } from '../styles/base';

export const Card = styled(Root)`
  background: #fff;
  display: table;
  font-family: sans-serif;
  border-radius: 3px;
  cursor: pointer;
  line-height: normal;
  position: relative;
  box-shadow: 0 1px 1px rgba(9, 30, 66, 0.2), 0 0 1px 0 rgba(9, 30, 66, 0.24);
  
  .wrapper {
    background: #E5E8EC;
    display: block;
    height: inherit;
    overflow: hidden;
    position: relative;
    border-radius: 3px;

    .img-wrapper{
      position: relative;
      width: inherit;
      height: inherit;
      display: block;

      img {
        position: absolute;
        top: 50%;
        left: 50%;
        max-height: 100%;
        max-width: 100%;
        display: block;
        transform: translate(-50%, -50%);
      }
    }
  }
`;
