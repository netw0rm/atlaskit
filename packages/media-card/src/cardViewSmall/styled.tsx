/* tslint:disable:variable-name */
import styled from 'styled-components';
import { Root, size, center, centerX, antialiased, ellipsis } from '../styles/base';
import {
  akColorN20,
  akColorN30,
  akColorN70
} from '@atlaskit/util-shared-styles';

const imageBackground = 'background: url("data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAoAAAAKAQMAAAC3/F3+AAAACXBIWXMAAAsTAAALEwEAmpwYAAAABlBMVEXf39////8zI3BgAAAAEUlEQVQIW2Nk38mIjH5wICMAez4Iyz2C/F8AAAAASUVORK5CYII=") repeat;';

export const SmallCard = styled(Root)`
  background-color: ${akColorN20};
  border-radius: 3px;
  font-family: sans-serif;
  cursor: pointer;
  box-sizing: border-box;
  height: 42px;
  padding: 5px;
  display: flex;
  align-items: center;
  
  &:hover {
    background-color: ${akColorN30};

    .title {
      color: #0065FF;
    }
  }

  .error-icon {
    height: 20px;
  }
`;

export const Retry = styled.div`
  ${antialiased()}
  ${ellipsis('calc(100%)')}
  font-weight: bold;
  color: #0065FF;
  font-size: 12px;
  line-height: 15px;
  margin-top: 2px;
`;

export const ImgWrapper = styled.div`
  ${center()}
  width: 32px;
  height: 100%;
  overflow: hidden;
  position: relative;
  
  img {
    ${imageBackground}
    max-width: 100%;
    max-height: 100%;   
  }
`;

export const Error = styled.div`
  ${antialiased()}
  ${ellipsis('calc(100%)')}
  font-weight: bold;
  color: ${akColorN70};
  font-size: 12px;
  line-height: 15px;
`;

export const Title = styled.div`
  ${antialiased()}
  ${ellipsis('calc(100%)')}
  font-weight: bold;
  color: #091E42;
  font-size: 12px;
  line-height: 15px;
`;

export const Size = styled.div`
  ${ellipsis('calc(100%)')}
  color: ${akColorN70};
  font-size: 12px;
  line-height: 15px;
  margin-top: 2px;
`;

export const RoundedBackground = styled.div`
  ${centerX()}
  min-width: 32px;
  height: inherit;
  background-color: #FAFBFC;
  border-radius: 3px;
  overflow: hidden;  
`;

export const InfoWrapper = styled.div`
  display: inline-block;
  padding-left: 8px;
  position: relative;
  max-width: calc(100% - 40px);
  vertical-align: middle;
`;

export const LoadingWrapper = styled.div`
  ${center()}
  color: #cfd4db;
  height: 100%;
`;

export const PlaceholderSmallWrapper = styled.div`
  ${size(32)}
  ${center()}
  position: relative;
  
  .file-type-icon {
    position: absolute;

    span {${size(12)}}
  }
`;
