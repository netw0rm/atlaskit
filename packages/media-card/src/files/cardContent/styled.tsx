/* tslint:disable:variable-name */
import styled from 'styled-components';

export const LoadingWrapper = styled.div`
  color: #cfd4db;
  display: flex;
  height: 100%;
  justify-content: center;
  align-items: center;
`;
