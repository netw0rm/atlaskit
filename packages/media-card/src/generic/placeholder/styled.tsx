/* tslint:disable:variable-name */
import styled from 'styled-components';

// MEDIA-TODO: Use an icon for &.fallback icon
export const PlaceholderWrapper = styled.div`
  display: flex;
  color: #cfd4db;
  height: 100%;
  justify-content: center;

  &.fallback {
    background: url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADAAAAAwCAAAAAByaaZbAAAAg0lEQVR4Ae2WsQ0DQQgEKXkb4FzGZZRBF/9l3FVAbuPMer/ESnbGBBCNVhuBxDJFEbUVstR3FNmuS8zjApKvipvoDoKtgtz1EKTAkALT4VcJ3aE7dIfuwCVMvJn10h+rhTthApMSkOOPCePIeTzKwjnwYpy3AgEv8IfdnBHc+OeEfX+eRtkEbLrqLcMAAAAASUVORK5CYII=) no-repeat center;
    background-size: initial;
    width: 100%;
    height: 100%;
  }
`;
