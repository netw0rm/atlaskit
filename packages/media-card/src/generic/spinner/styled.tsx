/* tslint:disable:variable-name */
import styled, { keyframes } from 'styled-components';

const l8 = keyframes`
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
`;

const spinnerLoad2 = keyframes`
  0% {
    transform: rotate(50deg);
  }
  100% {
    transform: rotate(230deg);
  }
`;

const spinnerLoad = keyframes`
  0% {
    transform: rotate(230deg);
  }
  100% {
    transform: rotate(510deg);
  }
`;

export const SpinnerWrapper = styled.div`
  position: absolute;
  z-index: 20;
  pointer-events: none;

  animation: ${spinnerLoad} 0.53s ease-in-out;
  animation-fill-mode: forwards;

  svg {
    display: block;
  }

  .path-animate {
    stroke-dasharray: 60;
    transform-origin: center;
    stroke: #42526e;

    stroke-dashoffset: 60;
    animation: ${l8} 0.86s infinite;
    animation-timing-function: cubic-bezier(0.4, 0.15, 0.6, 0.85);
    transition: stroke-dashoffset 0.8s ease-in-out, opacity 0.2s ease-in-out 0.45s;
    opacity: 0;
  }

  &.active {
    animation: ${spinnerLoad2} 1s ease-in-out;
    animation-fill-mode: forwards;

    .path-animate {
      stroke-dashoffset: 50;
      opacity: 1;
      transition: stroke-dashoffset 0.8s ease-in-out, opacity 0.2s ease-in-out;
    }
  }
`;

export default SpinnerWrapper;
