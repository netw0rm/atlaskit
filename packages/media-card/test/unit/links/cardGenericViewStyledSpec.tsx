/**
 * @jest-environment node
 */
import 'jest-styled-components';
import * as React from 'react';
import { shallow } from 'enzyme';
import { Thumbnail } from '../../../src/links/cardGenericView/styled';

describe('LinkCardGenericView styles', () => {
  it('Thumbnail should extend MediaImage and use default crop', () => {
    const card = shallow(<Thumbnail dataURI="" appearance="auto" />);

    expect(card).toMatchSnapshot();
  });
});
