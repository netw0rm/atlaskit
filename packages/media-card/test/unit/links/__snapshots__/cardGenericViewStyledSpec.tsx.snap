// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkCardGenericView styles Thumbnail should extend MediaImage and use default crop 1`] = `
<MediaImage
  appearance="auto"
  className="fPHjld"
  crop={true}
  dataURI=""
  fadeIn={true}
  height="100%"
  transparentFallback={false}
  width="100%"
/>
`;
